Fix account creation when the network is slow (#258)

marcinbodnar opened the pull request:
@vgrichina @kcole16 
With these changes, when an error occurs (on blockchain) while creating an account it will be still saved in Local Storage. There are two cases now:
1. If the account was created in blockchain but some error occurred, then the user will be able to continue with set recovery, like nothing happens.
2. If the account was not created in blockchain, it will be still saved in Local Storage, but immediately after this action (or anytime when user enters wallet) check is performed and the account will be removed from Local Storage.

With case 2. there might be a situation where user have other accounts, so I added additional functionality that will log him into another (first of list) account, so he will be able to continue with Wallet. Without that, user will not be able to login to different accounts until creating new one (because if user is not logged in, the list of accounts is not available).

I think we should also refine text message for case 2. for something like "The account is saved in Local Storage, but it doesn't exist on blockchain. The account was removed from Local Storage and you ware logged in to another account. You can try creating account with the same name again.". @kcole16 ?

kcole16 commented:
@marcinbodnar I think we should assume the majority of users are unfamiliar with local storage, which is why I kept the message pretty generic.

Though it is confusing that we show `Username is taken. Try something else.` when any error occurs during account creation.

marcinbodnar commented:
@kcole16 

> I think we should assume the majority of users are unfamiliar with local storage, which is why I kept the message pretty generic.

got it. I mostly worry that the user might not notice that was logged automatically into another account, and it might not be the same account that before.

> Though it is confusing that we show Username is taken. Try something else. when any error occurs during account creation.

I think I can overwrite it, what do you propose?

kcole16 commented:
> I think I can overwrite it, what do you propose?

Ideally, we would show the actual error message (I think this is covered in https://github.com/nearprotocol/near-wallet/issues/289).

In the meantime, we can display "An error occurred. Please try again!". 

We should also remove the red outline from the username input box, and enable the "Create Account" button. 

Important note: when the error is actually "Username is taken" we do still want to display that, highlight the input box, and disable "Create account".

marcinbodnar commented:
@kcole16 ok

vgrichina commented on src/utils/wallet.js:
```diff
@@ -208,11 +208,14 @@ export class Wallet {
+                newAccountPublicKey: keyPair.publicKey.toString()
+            })
+        } finally {
+            await this.saveAndSelectAccount(accountId, keyPair);
```
this should only be done only specifically for timeout errors (either network request timeout or neacore side timeout).

all other errors shouldn't try to use account as it's clearly not created

marcinbodnar commented:
@vgrichina I also implemented changes from PR#276 and added them here.

In `redirectToCreateAccount` we don't need `next_url` - redirection to the previous page is implemented with earlier changes and it's working fine.

`WALLET_CREATE_NEW_ACCOUNT_URL` stays in `wallet.js` because moving it to `Routing.js` cause dependencies issue - in this case, `Routing.js` will import from `actions/account.js` and `actions/account.js` will import from `Routing.js`.

@vgrichina @kcole16 Also, it seems that this functionality could be done a bit better, at this moment if account exists in local storage but does not exist in the blockchain it will always redirect user to Create Account page, even if there is another account available. I think it would be better to clear the account and if user has another account available we should leave him on the same page and show GlobalAlert with info, and if user does not have another account he will be redirected to Create Account anyway and we will show the same globalAlert. It's not very important change but I think it will be a bit better flow.

kcole16 commented:
@marcinbodnar Picking up this thread again. I think your proposed changes re: the flow makes sense (especially re: clearing the non-existent account from localStorage).

marcinbodnar commented:
@kcole16 ok, I will implement this flow, but it would be best to merge #390 first (@vgrichina ) because there are a lot of changes made, including some changes for this functionality.
